json-rpc/tests/integration_test.rs

```diff
@@ -858,6 +860,71 @@ fn create_test_cases() -> Vec<Test> {
+
+                    // We must check the transactions we got correspond to the hashes in the proofs
+                    let raw_blobs = data["serialized_transactions"].as_array().unwrap();
+                    assert!(!raw_blobs.is_empty());
+                    let actual_txs:Vec<Transaction>= raw_blobs.iter().map(|tx| {
+                        lcs::from_bytes(&hex::decode(&tx.as_str().unwrap()).unwrap()).unwrap()
+                    }).collect();
+                    assert!(!actual_txs.is_empty());
+                    assert_eq!(txs_infos.len(), actual_txs.len());
+                    for (index, tx) in actual_txs.iter().enumerate() {
+                        // Notice we need to actually hash the transaction to be sure its hash is correct
+                        assert_eq!(tx.hash(), txs_infos[index].transaction_hash());
+                    }
+
+                    // We compare our results with the non-veryfing API for the test
+                    let resp_tx = env.send("get_transactions", json!([base_version, txs_infos.len(), false]));
+                    let no_proof_txns = resp_tx.result.unwrap();
+                    assert!(!no_proof_txns.as_array().unwrap().is_empty());
+                    assert_eq!(no_proof_txns.as_array().unwrap().len(), actual_txs.len());
+                    for (index, tx) in no_proof_txns.as_array().unwrap().iter().enumerate() {
+                        assert_eq!(tx["hash"].as_str().unwrap(), actual_txs[index].hash().to_hex());
+                    }
+
+                    // We need to get the details required to verify the proof
+                    let li_raw = data["ledger_info"].as_str().unwrap();
+                    let li:LedgerInfoWithSignatures = lcs::from_bytes(&hex::decode(&li_raw).unwrap()).unwrap();
+                    let expected_hash = li.ledger_info().transaction_accumulator_hash();
+
+                    // We want to verify the signatures of the LedgerInfo to be sure it's valid, but
+                    // since we don't have a local state with the set of validators unlike the client,
+                    // we need to find a workaround to get the validator set. This works since it's a test
+                    // and the validator set is not changing, but it wouldn't work in practice.
+                    let state_data = env.send("get_state_proof", json!([base_version])).result.unwrap();
+                    let ep_cp = state_data["epoch_change_proof"].as_str().unwrap();
+                    let epoch_proofs:EpochChangeProof = lcs::from_bytes(&hex::decode(&ep_cp).unwrap()).unwrap();
+                    let latest_epoch = epoch_proofs.ledger_info_with_sigs.first().unwrap();
+                    assert!(li.verify_signatures(&latest_epoch.ledger_info().next_epoch_state().unwrap().verifier).is_ok());
+
+                    // and we eventually verify the proofs for the transactions
+                    assert!(li_to_tip.verify(expected_hash, Some(base_version), &hashes).is_ok());
```

Comment:
The expected_hash is also from server. Shouldn't we validate expected_hash first?

Reply:
also, you returned latest ledger info, but the hashes are from transactions base_version to limit, how could we use the latest ledger info for the case if limit is < ledger info.

Reply:
It's the case, expected_hash is validated by  `li.verify_signatures` in the assert above.

The latest ledger_info contains the necessary hash to prove the proofs that are generated at that ledger_info, but these proofs can be for any version < ledger_info

Reply:
one thing here, above "li.verify_signatures" validates signature, is it also validating transaction_accumulator_hash is the one expected?

Reply:
Yes